erpnext/controllers/accounts_controller.py

```diff
@@ -451,6 +451,7 @@ def clear_unallocated_advances(self, childtype, parentfield):
 	def apply_shipping_rule(self):
 		if self.shipping_rule:
 			shipping_rule = frappe.get_doc("Shipping Rule", self.shipping_rule)
+			shipping_rule.remove_existing_shipping_rule(self,"Shipping Rule", self.shipping_rule)
```

Comment:
- why are you giving self argument here ?

Reply:
Without self, I am not able to access doc to get taxes